pkg/apis/migration/v1alpha1/condition.go

```diff
@@ -203,6 +204,33 @@ func (r *Conditions) SetCondition(condition Condition) {
 	}
 }
 
+// Set (add/update) the specified condition to the collection.
+func (r *Conditions) SetAndLogCondition(logger logging.Logger, condition Condition, namespace, name string) {
```

Comment:
I'm not convinced we need to log setting the condition but instead could just log all of the `Conditions` object at the end of each reconcile.  That seems simpler and accomplishes the same thing.
That said:
- The `logr.Logger` (interface) should be used instead of `logging.Logger`.
- The logger should be configured at the beginning of the Reconcile() (near log.Reset()) with `, "storage", request)` so that all log entries has that kv pair instead of being passed in.
- Why log all of the condition fields separately?  Instead of just: `logger.Info("...", "condition", condition)`
- I don't think noting that a condition being set has already exists is useful.
- This method should not be re-implementing the SetCondition() logic.  Further, it is not doing it correctly.  The LastTransitionTime should only be updated when the condition has changed.

Reply:
+1. I agree with most points state here, will refactor.

> Further, it is not doing it correctly. The LastTransitionTime should only be updated when the condition has changed.

I am curious why should we not add LastTransitionTime to conditions that are being added? I interpret the transition time of the CR into that condition when the condition is added.